Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp

```diff
@@ -455,6 +471,664 @@ int currency_digits(StringView currency)
+                // 2. Append a new Record { [[Type]]: "compact", [[Value]]: compactName } as the last element of result.
+
+                // FIXME: Implement this when GetNotationSubPattern is fully implemented.
+            }
+
+            // vi. Else if p is equal to "scientificSeparator", then
+            else if (part == "scientificSeparator"sv) {
+                // 1. Let scientificSeparator be the ILND String representing the exponent separator.
+                // 2. Append a new Record { [[Type]]: "exponentSeparator", [[Value]]: scientificSeparator } as the last element of result.
+
+                // FIXME: Implement this when GetNotationSubPattern is fully implemented.
+            }
+
+            // vii. Else if p is equal to "scientificExponent", then
+            else if (part == "scientificExponent"sv) {
+                // 1. If exponent < 0, then
+                //     a. Let minusSignSymbol be the ILND String representing the minus sign.
+                //     b. Append a new Record { [[Type]]: "exponentMinusSign", [[Value]]: minusSignSymbol } as the last element of result.
+                //     c. Let exponent be -exponent.
+                // 2. Let exponentResult be ToRawFixed(exponent, 1, 0, 0).
+                // 3. Append a new Record { [[Type]]: "exponentInteger", [[Value]]: exponentResult.[[FormattedString]] } as the last element of result.
+
+                // FIXME: Implement this when GetNotationSubPattern is fully implemented.
+            }
+
+            // viii. Else,
+            else {
+                // 1. Let unknown be an ILND String based on x and p.
+                // 2. Append a new Record { [[Type]]: "unknown", [[Value]]: unknown } as the last element of result.
+
+                // FIXME: Implement this when GetNotationSubPattern is fully implemented.
+            }
+        }
+    }
+
+    // 5. Return result.
+    return result;
+}
+
+// 15.1.8 FormatNumeric ( numberFormat, x ), https://tc39.es/ecma402/#sec-formatnumber
+String format_numeric(NumberFormat& number_format, double number)
+{
+    // 1. Let parts be ? PartitionNumberPattern(numberFormat, x).
+    // Note: Our implementation of PartitionNumberPattern does not throw.
+    auto parts = partition_number_pattern(number_format, number);
+
+    // 2. Let result be the empty String.
+    StringBuilder result;
+
+    // 3. For each Record { [[Type]], [[Value]] } part in parts, do
+    for (auto& part : parts) {
+        // a. Set result to the string-concatenation of result and part.[[Value]].
+        result.append(move(part.value));
+    }
+
+    // 4. Return result.
+    return result.build();
+}
+
+static String cut_trailing_zeroes(StringView string, int cut)
+{
+    // These steps are exactly the same between ToRawPrecision and ToRawFixed.
+
+    // Repeat, while cut > 0 and the last character of m is "0",
+    while ((cut > 0) && (string.ends_with('0'))) {
```

Comment:
I'm already not saying anything about many of these extraneous parens, but in `(string.ends_with('0'))` they're completely useless :D

Reply:
Oops! Explicit parentheses everywhere is an old habit I picked up after too many bugs on really old legacy code. Around the `ends_with` call is definitely a mistake though.